lib/features/mailbox/presentation/model/mailbox_node.dart

```diff
@@ -32,4 +32,17 @@ class MailboxNode {
         ..add(node))
       .build();
   }
+
+  bool _validateChild(PresentationMailbox mailbox) {
+    if (mailbox.parentId == item.id) {
+      return true;
+    }
+    if (item == _root) {
```

Comment:
You can write more readable code.
````
return (mailbox.parentId == item.id || item == _root) ? true :  false;
````

Reply:
IMO, it quite complexes when combine conditionals.

Reply:
Agree, zero chance I understand this at first sight...